jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -106,53 +107,39 @@
      * @see SessionCache
      *
      */
-    public class SessionInactivityTimeout extends IdleTimeout
+    public class SessionInactivityTimeout extends CyclicTimeout
     {
-        /**
-         * 
-         */
+        protected long _timeout = -1;
+        
         public SessionInactivityTimeout()
         {
             super(getSessionHandler().getScheduler());
         }
 
-        /** 
-         * @see org.eclipse.jetty.io.IdleTimeout#onIdleExpired(java.util.concurrent.TimeoutException)
-         */
+
         @Override
-        protected void onIdleExpired(TimeoutException timeout)
+        public void onTimeoutExpired()
         {
             //called when the timer goes off
             if (LOG.isDebugEnabled()) LOG.debug("Timer expired for session {}", getId());
            getSessionHandler().sessionInactivityTimerExpired(Session.this);
         }
+        
 
-        /** 
-         * @see org.eclipse.jetty.io.IdleTimeout#isOpen()
-         */
-        @Override
-        public boolean isOpen()
+        public void setIdleTimeout(long ms)
         {
-            // Called to determine if the timer should be reset
-            // True if:
-            // 1. the session is still valid
-            // BUT if passivated out to disk, do we really want this timer to keep going off?
-            try (Lock lock = _lock.lock())
-            {
-                return isValid() && isResident();
-            }
+            if (LOG.isDebugEnabled()) LOG.debug("setIdleTimeout={}ms",ms);
+            _timeout = ms;
         }
 
-        /** 
-         * @see org.eclipse.jetty.io.IdleTimeout#setIdleTimeout(long)
-         */
-        @Override
-        public void setIdleTimeout(long idleTimeout)
+        public void start ()
```

Comment:
if you called this `schedule` then I wouldn't ask you to make this class a `LifeCycle`!